mp/src/game/server/filters.cpp

```diff
@@ -372,7 +372,7 @@ class FilterDamageType : public CBaseFilter
 
 	bool PassesDamageFilterImpl(const CTakeDamageInfo &info)
 	{
-	 	return info.GetDamageType() == m_iDamageType;
+		return ((info.GetDamageType() & m_iDamageType) == m_iDamageType);
```

Comment:
Random bug fix that I found in a fork from Tony. He said this fixed issues with more complex damage types (as this wasn't properly doing a bitwise-check before)